tests/integration/backward_compatible/proxy/map_nearcache_test.py

```diff
@@ -108,3 +111,71 @@ def fill_map_and_near_cache(self, count=10):
         for k, v in six.iteritems(fill_content):
             self.map.get(k)
         return fill_content
+
+
+ENTRY_COUNT = 100
+
+
+class NonStopNearCacheTest(HazelcastTestCase):
+    def setUp(self):
+        rc = self.create_rc()
+        cluster = self.create_cluster(rc, self.read_cluster_config())
+        cluster.start_member()
+
+        def event_collector():
+            events = []
+
+            def collector(e):
+                if e == LifecycleState.DISCONNECTED:
+                    events.append(e)
+
+            collector.events = events
+            return collector
+
+        collector = event_collector()
+
+        client = HazelcastClient(
+            cluster_name=cluster.id,
+            reconnect_mode=ReconnectMode.ASYNC,
+            near_caches={"map": {}},
+            lifecycle_listeners=[collector],
+        )
+
+        map = client.get_map("map").blocking()
+
+        for i in range(ENTRY_COUNT):
+            map.put(i, i)
+
+        # Populate the near cache
+        for i in range(ENTRY_COUNT):
+            map.get(i)
+
+        rc.terminateCluster(cluster.id)
+        rc.exit()
+
+        self.assertTrueEventually(lambda: self.assertEqual(1, len(collector.events)))
```

Comment:
Is there any possibility for the disconnected event to fire more than once? If so, the test will be flaky

Reply:
Nope, the client will disconnect only once